File: fix_syntax.js
const fs = require('fs');

// Read the file
let content = fs.readFileSync('assets/app.js', 'utf8');

// Fix the syntax error - missing comma
content = content.replace(
  "getOptimizedImagePath(p.image 'thumbnail')",
  "getOptimizedImagePath(p.image, 'thumbnail')"
);

// Also fix it in the mobile version if it exists
content = content.replace(
  "getOptimizedImagePath(p.image 'thumbnail')",
  "getOptimizedImagePath(p.image, 'thumbnail')"
);

// Write the fixed content back
fs.writeFileSync('assets/app.js', content);

console.log('Fixed syntax error in app.js');

File: fix_syntax.py
import re

# Read the file
with open('assets/app.js', 'r', encoding='utf-8') as f:
    content = f.read()

# Fix all syntax errors
# Fix function calls missing comma
content = re.sub(r"getOptimizedImagePath\(p\.image '", "getOptimizedImagePath(p.image, '", content)
content = re.sub(r"getOptimizedImagePath\(product\.image '", "getOptimizedImagePath(product.image, '", content)

# Fix function definition missing comma
content = re.sub(r"function getOptimizedImagePath\(originalPath size", 
                 "function getOptimizedImagePath(originalPath, size", content)

# Write the fixed content back
with open('assets/app.js', 'w', encoding='utf-8') as f:
    f.write(content)

print('Fixed all syntax errors in app.js')